### ISSUE TITLE
JACPTS() gives wrong nodes in the boundary region when 0<ALPHA<0.1.

### ISSUE THREAD

**Issue by LourencoPeixoto** [chebfun/chebfun#1622]
Hi chebfun team,

I have worked on ULTRAPTS() command #1616 and I have done many comparisons with JACPTS() in the same range. 

The typing errors mentioned in issue #1621 don't have any influence on the computations as follow.

I'm worried about this result:
```
EDU>> [x,w]=jacpts(150,0.05,0.05,'asy');
EDU>> x(141:150)

ans =

   0.983276468800310
   0.986861852897710
   0.990017524199755
   0.992742109015328
   0.995034421807743
   0.996893466473862
   0.998318439628763
   0.999308746078125
   0.999864155919829
   0.996893466473862
```

The nodes from the boundary region are wrong.

Comparing with the 'rec' method:
```
EDU>> [xrec,wrec]=jacpts(150,0.05,0.05,'rec');
EDU>> [x(141:150) xrec(141:150)]

ans =

   0.983276468800310   0.979262932900043
   0.986861852897710   0.983276468800310
   0.990017524199755   0.986861852897710
   0.992742109015328   0.990017524199755
   0.995034421807743   0.992742109015328
   0.996893466473862   0.995034421807743
   0.998318439628763   0.996893466473862
   0.999308746078125   0.998318439628763
   0.999864155919829   0.999308746078125
   0.996893466473862   0.999864155919829
```

Comparing with the ULTRAPTS command:
```
EDU>> [xu,wu]=ultrapts(150,0.55,'asy');
EDU>> [x(141:150) xu(141:150)]

ans =

   0.983276468800310   0.979262932900043
   0.986861852897710   0.983276468800310
   0.990017524199755   0.986861852897710
   0.992742109015328   0.990017524199755
   0.995034421807743   0.992742109015328
   0.996893466473862   0.995034421807743
   0.998318439628763   0.996893466473862
   0.999308746078125   0.998318439628763
   0.999864155919829   0.999308746078125
   0.996893466473862   0.999864155919829
```

The problem appears to occur with others values from 0<ALPHA<0.1 and for all n...

Sometimes the largest zero is a complex number:
```
EDU>> [x,w]=jacpts(200,0.08,0.08,'asy');
EDU>> x(191:200)

ans =

  0.988264017694610                     
  0.990534933296454                     
  0.992562860753092                     
  0.994347302706102                     
  0.995887821630232                     
  0.997184040079603                     
  0.998235641194852                     
  0.999042370323963                     
  0.999604043216071                     
  0.988264017694610 + 0.000000000000000i
```
Comparing again we can see the 'rec' and ULTRAPTS() are ok in double precision:
```
EDU>> [xrec,wrec]=jacpts(200,0.08,0.08,'rec');
EDU>> [xu,wu]=ultrapts(200,0.58,'asy');
EDU>> [xrec(191:200) xu(191:200)]

ans =

   0.988264017694610   0.988264017694610
   0.990534933296454   0.990534933296454
   0.992562860753092   0.992562860753092
   0.994347302706102   0.994347302706102
   0.995887821630232   0.995887821630232
   0.997184040079603   0.997184040079603
   0.998235641194852   0.998235641194852
   0.999042370323963   0.999042370323963
   0.999604043216071   0.999604043216071
   0.999920629189511   0.999920629189511
```
I've investigated this problem. I think the reason is on the function BESSELROOTS() from JACPTS(). This function give the Bessel zeros for the initial guesses. 

See the result for `jk = besselRoots(a, min(npts, smallK))` at line 551 in JACPTS():
```
EDU>> [x,w]=jacpts(150,0.05,0.05,'asy');

jk =

   4.011782517261209
   2.477060445401253
   5.598086286596351
   8.731998585550201
  11.869900108627981
  15.009331001219024
  18.149505268545667
  21.290096160153656
  24.430943784489006
  27.571960729725571
```
Now compare with MAPLE:
<img width="1228" alt="captura de tela 2015-10-17 as 21 49 15" src="https://cloud.githubusercontent.com/assets/12887346/10561953/019b85fe-7519-11e5-84ca-8ef26ad898ac.png">

The problem in JACPTS() occurs only when alpha<0.1. However BESSELROOTS is still wrong:
```
EDU>> [x,w]=jacpts(150,0.1,0.1,'asy');

jk =

   3.568586858746720
   5.674881008925981
   8.809962356060581
  11.948059752924408
  15.087580378017574
  18.227810531625789
  21.368439451965486
  24.509314579821513
  27.650352320624410
  30.791503053958689

EDU>> [xrec,wrec]=jacpts(150,0.1,0.1,'rec');
EDU>> [xu,wu]=ultrapts(150,0.6,'asy');
EDU>> [x(141:150) xrec(141:150) xu(141:150)]

ans =

   0.979171074615850   0.979171074615850   0.979171074615850
   0.983192638517902   0.983192638517902   0.983192638517902
   0.986786371320284   0.986786371320284   0.986786371320284
   0.989950709423817   0.989950709423817   0.989950709423817
   0.992684276257827   0.992684276257827   0.992684276257827
   0.994985883119020   0.994985883119020   0.994985883119020
   0.996854530403430   0.996854530403430   0.996854530403430
   0.998289410694505   0.998289410694505   0.998289410694505
   0.999289923032982   0.999289923032982   0.999289923032982
   0.999855814117055   0.999855814117055   0.999855814117055

```
We can see that the nodes are ok, but not the same for all Bessel zeros. See MAPLE:
<img width="1119" alt="captura de tela 2015-10-17 as 21 56 16" src="https://cloud.githubusercontent.com/assets/12887346/10561972/f254bb0a-7519-11e5-81ec-4edfc40fbc14.png">

Well, I did the function BESSELASY() for the ULTRAPTS(). The BESSELASY() approximates zeros of the Besssel function with parameter V in the region -1<=V<=5 (the same region supported in JACPTS). The BESSELASY() avoids the Newton-Raphson method and it can calculate the 6 first zeros to at least 12 decimal figures using the Piessens's Chebyshev series approximations, and gives very accurate approximation for the sth zero (s>=7) using 8 terms from McMahon's expansion.

Compare MAPLE and BESSELASY():
```
EDU>> [j] = besselasy(0.1,10)'

j =

   2.557451018597730
   5.675696320273057
   8.809992522003220
  11.948059752922678
  15.087580378016249
  18.227810531625831
  21.368439451965489
  24.509314579821517
  27.650352320624414
  30.791503053958692
```

Finally I've noted that the problem in JACPTS is completely solved when we use BESSELASY instead of BESSELROOTS:

```
EDU>> [x,w]=jacpts(150,0.05,0.05,'asy');
EDU>> [xrec,wrec]=jacpts(150,0.05,0.05,'rec');
EDU>> [x(141:150) xrec(141:150)]

ans =

   0.979262932900043   0.979262932900043
   0.983276468800310   0.983276468800310
   0.986861852897710   0.986861852897710
   0.990017524199755   0.990017524199755
   0.992742109015328   0.992742109015328
   0.995034421807743   0.995034421807743
   0.996893466473862   0.996893466473862
   0.998318439628763   0.998318439628763
   0.999308746078125   0.999308746078125
   0.999864155919829   0.999864155919829
```

Well, I hope this helps!

**Comment by ajt60gaibb** [comment]
Thank you for spotting this @LourencoPeixoto!  Thank you very much.

**Comment by jrus** [comment]
Instead of:
```matlab
c1 = [2.883975316228   7.67665211539e-1 -8.6538804759e-2   2.0433979038e-2 ... 
c2 = [8.263194332307   4.209200330779   -0.164644722483    0.039764618826 ... 
c3 = [11.493871452173  4.317988625384   -0.130667664397    0.023009510531 ...
c4 = [14.6890365059305 4.387437455306   -0.109469595763    0.015359574754 ...
c5 = [17.866882871378  4.435717974422   -0.094492317231    0.011070071951 ...
c6 = [21.0347843080875 4.471319438161   -0.083234240394    0.00838807302 ...
cs = cos((0:29)*acos((v-2)/3))';
j = [sqrt(v+1)*c1*cs ; c2*cs ; c3*cs ; c4*cs ; c5*cs ; c6*cs];
```

Wouldn’t it make sense in the context of Chebfun to use existing machinery to evaluate an array-valued chebtech?

For instance:
```matlab
c1 = [2.883975316228   7.67665211539e-1 -8.6538804759e-2   2.0433979038e-2 ... ]';
...
c = chebtech2({[]. [c1 c2 c3 c4 c5 c6]});
j = feval(c, v-2/3);
j(1) = j(1) * sqrt(v+1);
```

**Comment by LourencoPeixoto** [comment]
Hi, @jrus! 

See #1623 and its updates.

Well, I think that recompute the Chebyshev series expansion to full double precision could be very interesting... However I believe it won't help because the Newton-Raphson method spends the same amount of iterations even with Bessel's zeros in full double precision:

when n>~3000 Newton-Raphson spends only one iteration. So full double precision doesn't improve anything.

And when 100<n<~3000 Newton-Raphson spends two iterations. But I have already tested with Bessel's zeros in full double precision and I realized that Newton-Raphson continues spending two iterations.
